### ISSUE TITLE
[iOS|Android] WebView events delegation

### ISSUE THREAD

**Issue by sandervandenbrink** [ionic-team/capacitor#1298]
In Android the Bridge class and in iOS the Bridge View Controller don't delegate the WebView events. 
I'm using Capacitor to fetch the web app remotely and if loading of the client fails (e.g. timeout or host not found) then there is no way to handle this with for example a screen that informs the user.

Perhaps an idea would be to have a listener/delegate property available in the Android Bridge class and the iOS CAPBridgeViewController that can optionally be set before the webview is being loaded.

Additionally I noticed that the WKWebView reload won't work if the initial load of the web client failed (because the url will be nil) so it would be convenient if the CAPBridgeViewController has a public load/reload method that uses the url from the config.

I don't have any other use cases than the error methods [webView(_:didFailProvisionalNavigation:withError:)](https://developer.apple.com/documentation/webkit/wknavigationdelegate/1455637-webview) and [onReceivedError](https://developer.android.com/reference/android/webkit/WebViewClient.html#onReceivedError(android.webkit.WebView,%20int,%20java.lang.String,%20java.lang.String)) (Note that this one is deprecated since API 23), so I'm not sure whether it's necessary to delegate more or all the webview events.

**Comment by mlynch** [comment]
Makes sense. We can add this to our list of tasks for making Capacitor work well embedded in other apps (where you're managing the instance of the bridge)

**Comment by eric-horodyski** [comment]
Closing as part of our backlog cleanup effort to keep Capacitor focused on current, actionable work.

More context: [https://ionic.io/blog/keeping-capacitors-backlog-healthy](https://ionic.io/blog/keeping-capacitors-backlog-healthy)

If this is still reproducible on the latest version, please share updated details and we can revisit.
